lib/include/openamp/rpmsg_virtio.h

```diff
@@ -41,9 +41,27 @@ struct rpmsg_virtio_shm_pool {
 	size_t size;
 };
 
+/**
+ * struct rpmsg_virtio_config - configuration of rpmsg device based on virtio
+ * @txbuf_size: the tx buffer size, used to send data from host to remote
+ * @rxbuf_size: the rx buffer size, used to send data from remote to host
```

Comment:
Deeply reviewing your patch, look to me that, for an unfamiliar user, it is not easy to understand this structure. 
My apologize for that because I should see it before... only the host use the config structure.
Another point is the notion of RX TX which is ambiguous. If only the host defines it, the remote gets the corresponding  values from the vrings.

Here is a proposal to rework the structure

```
 * struct rpmsg_virtio_config - configuration of rpmsg device based on virtio
 *
 * This structure is use by the rpmsg virtio host to configure the virtiio layer.
 * 
 * @h2r_buf_size: the buffer size, used to send data from host to remote
 * @r2h_buf_size: the uffer size, used to send data from remote to host
 struct rpmsg_virtio_config {
	uint32_t h2r_buf_size;
	uint32_t r2h_buf_size;
}
```